fix: solves display bug of category chevron buttons

abhiram11 opened the pull request:
## Description

Closes #628 

- [x] I read the [contributing docs](CONTRIBUTING.md) (if this is your first contribution)

## Type of change

<!--- Please delete options that are not relevant. -->

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

Note: Kindly check the Bug Report #628 where I have elaborated on the issue in more detail.

This PR solves the display bug where the `<` and `>` category buttons require an extra event to be called/clicked in order to render on the page as required by the logic.

## This is how the proposed solution in my PR looks like:
**The URL shows the difference between the current behavior (lenstube) and the expected behavior (localhost)**

https://user-images.githubusercontent.com/20269286/219801637-929afbb0-0477-4449-b05b-8a4f7b84fa75.mp4



### P.S. I am new to open-source contributions, I am all ears to feedback! Furthermore, we can also refactor the filename `utils/hooks/useHorizantalScroll` and correct the spelling to `Horizontal`. ^^

sasicodes commented on apps/web/src/components/Common/CategoryFilters.tsx:
```diff
@@ -38,14 +38,31 @@ const CategoryFilters = () => {
     Analytics.track(TRACK.CLICK_CATEGORIES_SCROLL_BUTTON)
+
     if (scrollRef.current) {
+      /**
```
Great work!

Thanks for the comments to convey. We can clean up the comments 👍🏻

abhiram11 replied:
Thanks for the feedback, I will remove the comments and set the variable type to `const` in the next PR.

Should I change the filename in this one itself or make a new PR for it?

sasicodes replied:
Update all comments in this same PR!

sasicodes commented on apps/web/src/components/Common/CategoryFilters.tsx:
```diff
@@ -38,14 +38,31 @@ const CategoryFilters = () => {
+       */
+
+      // That's why I used a new variable to handle the calculation of "shift" effectively
+      let scrolled = scrollRef.current.scrollLeft + shift
```
can be `const` if we are not reassigning the value 🙇🏻

abhiram11 replied:
Sure! Let me update that, I overlooked the type after the whole debugging I guess.

sasicodes commented:
@abhiram11 We can rename typo in the the file name to correct spelling 👍🏻

abhiram11 commented:
I have pushed a new commit with the following updates:
1. Changed `let` to `const`
2. Removed Comments
3. Updated filename to `useHorizontalScroll` and modified wherever it is called.

P.S. I am new to open source contribution and git, so if there are any issues with my PR or git calls, please provide feedback for the same!

sasicodes commented:
> I have pushed a new commit with the following updates:
> 
> 1. Changed `let` to `const`
> 2. Removed Comments
> 3. Updated filename to `useHorizontalScroll` and modified wherever it is called.
> 
> P.S. I am new to open source contribution and git, so if there are any issues with my PR or git calls, please provide feedback for the same!

You did all great. Thank you for your contribution. 😊

abhiram11 commented:
Thank you for the feedback! ^^ Feel free to let me know if there are any other issues that I can help with, especially in Javascript, Python, and Frontend.

Have a great weekend!

sasicodes reviewed (approved):
![Dog LGTM Keyboard](https://media.giphy.com/media/kbVzENVNlE60MQR1HY/giphy.gif)

sasicodes commented:
> 

You can always pick from issues or suggest feature requests (roadmap.lenstube.xyz)

sasicodes commented:
@abhiram11 ☝🏻

abhiram11 commented:
This is great! Thanks for the recognition!! 😊
